execution_engine/src/storage/global_state/mod.rs

```diff
@@ -137,26 +535,23 @@ where
     S::Error: From<R::Error>,
     E: From<R::Error> + From<S::Error> + From<bytesrepr::Error> + From<CommitError>,
 {
-    let mut txn = environment.create_read_write_txn()?;
-    let mut state_root = prestate_hash;
-    let maybe_root: Option<Trie<Key, StoredValue>> = store.get(&txn, &state_root)?;
-    if maybe_root.is_none() {
-        return Err(CommitError::RootNotFound(prestate_hash).into());
+    let txn = environment.create_read_txn()?;
+    let state_root = prestate_hash;
+    let mut fancy_trie = fancy_trie::Pointer {
+        link: fancy_trie::Link::GlobalState(state_root),
+        // This value doesn't matter because:
+        // * If nothing is inserted into the fancy trie, then nothing will be written to global
+        //   state and this value won't be used.
+        // * If something is put into the fancy trie, it will trigger a global state read without
+        //   consulting this value and it will then be replaced based on the data read from
+        //   global state.
+        is_leaf: false,
```

Comment:
Perhaps `Option<bool>` makes sense to represent a state that doesn't matter? Or turning `fancy_trie::Pointer` would make this representation easier to represent?

Reply:
So, to clarify, that field does matter, it's just that whichever value we put on this particular instance doesn't matter for the reasons stated in the comment. Making it `Option<bool>` would introduce unnecessary overhead. Not sure what you mean by “turning `fancy_trie::Pointer`”.

Reply:
> Not sure what you mean by “turning `fancy_trie::Pointer`”.

Into an enum

Reply:
Oh, we have `Link<K, V>` for that. Since inserting can never change the status of a trie from leaf to non-leaf or vice-versa, I can change the definition of `insert` to take a `Link` instead. Nice catch! This design is a residue from a previous recursive implementation.

Reply:
Nevermind, we do need the full `Pointer` to load the global state entry on top of.